### PR TITLE
v1.1.0 Fix multiple R codes in Clarity extract

### LINKED ISSUE #24: Multiple R codes in Clarity extract breaks things
When using a Clarity extract with `--clarity_extract` and `--use_paths`, if a sample has multiple R codes like `R208.1|R208.2` in the Test Directory Test Code field this seems to break things:
```
    main()
  File "/app/pandora.py", line 291, in main
    clarity_handler.handle_clarity_extract(
  File "/app/utils/clarity_extract_handler.py", line 538, in handle_clarity_extract
    report_df = filtering_reports(report_df)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/utils/clarity_extract_handler.py", line 358, in filtering_reports
    report_df["rcode_match"] = report_df.apply(is_report_code_in_list, axis=1)
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/frame.py", line 10381, in apply
    return op.apply().__finalize__(self, method="apply")
           ^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/apply.py", line 916, in apply
    return self.apply_standard()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/apply.py", line 1063, in apply_standard
    results, res_index = self.apply_series_generator()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/apply.py", line 1081, in apply_series_generator
    results[i] = self.func(v, *self.args, **self.kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/utils/clarity_extract_handler.py", line 292, in is_report_code_in_list
    if pd.isna(r_codes) or r_codes == [] or r_codes == "[]":
       ^^^^^^^^^^^^^^^^
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

### PR DESCRIPTION
**Pull request by rklocke**
- Add new function to pre-process Clarity extract to handle cases where multiple of the same R code but different test modes, and remove rows with multiple unique base R codes as these are not being handled in this version (fixes #24 )
  - Add unit tests
  - Only process Clarity extract if rows remain after preprocessing
- Change order of things to be able to report missing data in DNAnexus, as these were previously silently removed when checking if R code in file from DNAnexus is in list of R codes from Clarity
- Remove unused function `find_file_name()` and unit tests for this function

- - -
This change is [Reviewable](https://reviewable.io/reviews/eastgenomics/clinvar_submissions/25)

## Summary by CodeRabbit

* **Bug Fixes**
  * Skip processing and outputs when clarity extract is empty to avoid misleading files and errors.
  * More robust handling of code comparisons and missing values across flows.

* **Improvements**
  * Outputs separate clarity-issues CSV when issues are found and includes it in downstream reporting.
  * Clearer separation of preprocessing, fetching, filtering and deduplication; filtering now preserves and returns missing and duplicate report info.

* **Tests**
  * Expanded coverage for preprocessing, filtering and edge cases; expectations updated for new return shapes.

### CODE REVIEW

**Review by RSWilson1 — CHANGES_REQUESTED** [review]
> @RSWilson1 reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/clinvar_submissions/25)* status: all files reviewed, 4 unresolved discussions (waiting on @rklocke)

___
*[`utils/clarity_extract_handler.py` line 70 at r2](https://reviewable.io/reviews/eastgenomics/clinvar_submissions/25#-Od8V4wzDKLzjEzhlphI:-Od8V4wzDKLzjEzhlphJ:b-uw78li) ([raw file](https://github.com/eastgenomics/clinvar_submissions/blob/5db237f9068de607653fea0694632c0d817f4029/utils/clarity_extract_handler.py#L70)):*
> ```Python
>     multiple_r_codes_df = clarity_df[clarity_df["R_codes"].str.len() > 1].copy()
> 
>     if not missing_r_codes_df.empty:
> ```

If theses are just printed out we don't have a record that these haven't been processes and any corrections required. Also, this removes all samples with multiple R_codes, which might exclude a lot of samples.
___
*[`utils/clarity_extract_handler.py` line 206 at r2](https://reviewable.io/reviews/eastgenomics/clinvar_submissions/25#-Od8ZiYEDGeCUcxkGi1F:-Od8ZiYEDGeCUcxkGi1G:b-l7dbor) ([raw file](https://github.com/eastgenomics/clinvar_submissions/blob/5db237f9068de607653fea0694632c0d817f4029/utils/clarity_extract_handler.py#L206)):*
> ```Python
>         futures = {
>             executor.submit(query_reports_for_project, pid, chunk): (
>                 pid,
> ```

not sure on using pid here as makes me think of personal identifiable data maybe p_id? or proj_id?

File: tests/test_clarity_extract_handler.py [hunk 1]
```diff
@@ -316,3 +260,256 @@ def test_case_sensitive(self, example_df_lowercase_r_codes):
...
+        expected_missing_data_df = pd.DataFrame(
+            {
+                "file_name": [pd.NA],
+                "sample_id": ["24123R0015"],
+                "report_r_code": [pd.NA],
+                "R_codes": [["R228"]],
+            }
+        )
+
+        pd.testing.assert_frame_equal(missing_data_df, expected_missing_data_df)
```
[the remark this answers is not in the document]
Comment (rklocke) [reply tests/test_clarity_extract_handler.py:585]: fixed, unit tests pass now

File: utils/clarity_extract_handler.py [hunk 2]
```diff
@@ -37,6 +37,50 @@ def open_files(clarity):
...
+    Inputs:
+        clarity_df (pd.DataFrame): DataFrame containing clarity extract information.
+    Outputs:
+        clarity_df (pd.DataFrame): Preprocessed DataFrame with additional
+        sample_id and R_codes columns.
+    """
+    # Split out sample ID to remove "SP-" prefix
+    clarity_df["sample_id"] = (
+        clarity_df["Specimen Identifier"].str.split("-").str[1]
+    )
```
[the remark this answers is not in the document]
Comment (rklocke) [reply utils/clarity_extract_handler.py:56]: Done

File: utils/clarity_extract_handler.py [hunk 3]
```diff
@@ -355,85 +375,86 @@ def filtering_reports(report_df):
...
-    ]
+    if "file_name" in report_df.columns:
+        filtered_df = report_df[
+            ~report_df["file_name"].str.contains(r"_(CNV|mosaic)_", na=False)
+        ].copy()
+    else:
+        filtered_df = report_df.copy()
 
-    return filtered_df
+    return filtered_df, missing_data_df
```
[the remark this answers is not in the document]
Comment (rklocke) [reply utils/clarity_extract_handler.py:418]: This was fixed

**Review by rklocke — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/clinvar_submissions/25)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @RSWilson1)

___
*[`utils/clarity_extract_handler.py` line 70 at r2](https://reviewable.io/reviews/eastgenomics/clinvar_submissions/25#-Od8V4wzDKLzjEzhlphI:-Od8qXbW0ziub-hq2TGN:b-3zxovw) ([raw file](https://github.com/eastgenomics/clinvar_submissions/blob/5db237f9068de607653fea0694632c0d817f4029/utils/clarity_extract_handler.py#L70)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

If theses are just printed out we don't have a record that these haven't been processes and any corrections required. Also, this removes all samples with multiple R_codes, which might exclude a lot of samples.
</blockquote></details>

Done, these should now be outputted to CSV with the other issue dfs
___
*[`utils/clarity_extract_handler.py` line 206 at r2](https://reviewable.io/reviews/eastgenomics/clinvar_submissions/25#-Od8ZiYEDGeCUcxkGi1F:-Od8qi5e-MUjr4naJYYD:bb6un1j) ([raw file](https://github.com/eastgenomics/clinvar_submissions/blob/5db237f9068de607653fea0694632c0d817f4029/utils/clarity_extract_handler.py#L206)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

not sure on using pid here as makes me think of personal identifiable data maybe p_id? or proj_id?
</blockquote></details>

Done

**Review by RSWilson1 — APPROVED** [review]
> @RSWilson1 reviewed 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/clinvar_submissions/25)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @rklocke)
